pkg/cmd/pr/shared/editable_http.go

```diff
@@ -58,7 +58,44 @@ func UpdateIssue(httpClient *http.Client, repo ghrepo.Interface, id string, isPR
 		})
 	}
 
-	if dirtyExcludingLabels(options) {
+	if options.Assignees.Edited {
+		client := api.NewClientFromHTTP(httpClient)
+		meReplacer := NewMeReplacer(client, repo.RepoHost())
+
+		if len(options.Assignees.Add) > 0 {
+			wg.Go(func() error {
+				addLogins, err := meReplacer.ReplaceSlice(options.Assignees.Add)
+				if err != nil {
+					return err
+				}
+
+				addIDs, err := options.Metadata.MembersToIDs(addLogins)
+				if err != nil {
+					return err
+				}
+
+				return addAssignees(httpClient, repo, id, addIDs)
+			})
+		}
+
+		if len(options.Assignees.Remove) > 0 {
+			wg.Go(func() error {
+				removeLogins, err := meReplacer.ReplaceSlice(options.Assignees.Remove)
+				if err != nil {
+					return err
+				}
+
+				removeIDs, err := options.Metadata.MembersToIDs(removeLogins)
+				if err != nil {
+					return err
+				}
+
+				return removeAssignees(httpClient, repo, id, removeIDs)
+			})
+		}
```

Comment:
This logic is dependent on `ReplaceValue()` logic removing editable content from list of content to be added, allowing it to run concurrently without having the same content being added and removed, correct?

Reply:
the `ReplaceValue()` is **only** for interaction mode, where we get a list of selected users which may contain deselected users from the default state. The selected list will be compared to the default list and reduced to have correct values in the `Add` and `Remove` fields before we do the queries.

the `ReplaceValue()` is **not** used when we run the cli with `--add-assignee` and `--remove-assignee` flags, we handle both flags in their own goroutine and let the graphql mutation handle the data change. 

this could introduce an edge case where `--add-assignee wingleung --remove-assignee wingleung` is a race condition with 2 possible outcomes.

Alternative would be to combine multiple mutations in 1 query, which would ensure the mutations run in order [ℹ️](https://graphql.org/learn/queries/#multiple-fields-in-mutations). 

However, another edge case would then be `--remove-assignee wingleung --add-assignee wingleung`. 
We don't know the order of which the flags were given, so if we always do the `addAssigneesToAssignable` first and then the `removeAssigneesFromAssignable` then the last flag in our edge case will not do it's job.

I'm overthinking it, the user of the cli is responsible of using correct flag values, `--add-assignee wingleung --remove-assignee wingleung` is prolly not a real life use case. And afterwards we just need to pass it on to graphql as fast as we can 😄